libbeat/processors/actions/alterFieldProcessor.go

```diff
@@ -0,0 +1,128 @@
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/pkg/errors"
+)
+
+// alterFieldNameFunc defines how fields must be processed
+type alterFieldFunc func(field string) string
+
+type alterFieldProcessor struct {
+	Fields        []string
+	IgnoreMissing bool
+	FailOnError   bool
+	FullPath      bool
+
+	processorName string
+	alterFunc     alterFieldFunc
+}
+
+// NewAlterFieldNameProcessor is an umbrella method for processing events based on provided fields. Such as converting event keys to uppercase/lowercase
+func NewAlterFieldProcessor(c *conf.C, processorName string, changeFunc alterFieldFunc) (processors.Processor, error) {
+	config := struct {
+		Fields        []string `config:"fields"`
+		IgnoreMissing bool     `config:"ignore_missing"`
+		FailOnError   bool     `config:"fail_on_error"`
+		FullPath      bool     `config:"fail_path"`
+	}{
+		IgnoreMissing: false,
+		FailOnError:   true,
+		FullPath:      false,
+	}
+
+	if err := c.Unpack(&config); err != nil {
+		return nil, fmt.Errorf("failed to unpack the %s fields configuration: %s", processorName, err)
+	}
+
+	// Skip mandatory fields
+	for _, readOnly := range processors.MandatoryExportedFields {
+		for i, field := range config.Fields {
+			if field == readOnly {
+				config.Fields = append(config.Fields[:i], config.Fields[i+1:]...)
+			}
+		}
+	}
+
+	return &alterFieldProcessor{
+		Fields:        config.Fields,
+		IgnoreMissing: config.IgnoreMissing,
+		FailOnError:   config.FailOnError,
+		FullPath:      config.FullPath,
+		processorName: processorName,
+		alterFunc:     changeFunc,
+	}, nil
+
+}
+
+func (a *alterFieldProcessor) String() string {
+	return fmt.Sprintf("%s fields=%+v", a.processorName, *a)
+}
+
+func (a *alterFieldProcessor) Run(event *beat.Event) (*beat.Event, error) {
+	var backup *beat.Event
+	if a.FailOnError {
+		backup = event.Clone()
+	}
+
+	for _, field := range a.Fields {
+		err := a.alter(event, field)
+		if err != nil {
+			if a.FailOnError {
+				event = backup
+				event.PutValue("error.message", err.Error())
+				return event, err
+			}
+		}
+	}
+
+	return event, nil
+}
+
+func (a *alterFieldProcessor) alter(event *beat.Event, field string) error {
+	// Get the value of the field to alter
+	value, err := event.GetValue(field)
+	if err != nil {
+		if a.IgnoreMissing && errors.Is(err, mapstr.ErrKeyNotFound) {
+			return nil
+		}
+		return fmt.Errorf("could not fetch value for key: %s, Error: %v", field, err)
+	}
+
+	// Delete the exisiting value
+	if strings.ContainsRune(field, '.') && a.FullPath {
+		// In case of full_path set to true, we need to make sure to modify the all keys in the chain
+		firstField := field[:strings.Index(field, ".")]
+		if err := event.Delete(firstField); err != nil {
+			return fmt.Errorf("could not delete key: %s, Error: %v", field, err)
+		}
```

Comment:
This does not look right, this code would delete the root of the graph in the event and would lead to data loss. Consider this test case:

```
fiRst: {
  some: "data"
  secOnd: {
    other: "data"
    Third: "value"
  }
}

if you ran `lower_case("fiRst.secOnd.Third", full_path: true)` it would delete `some: "data` and `other: "data` and only keep `first.second.third: "value"`.

If we want to support `full_path` we must traverse the graph key by key and recursively replace each key on each level instead of deleting and replacing the root. I'm fine with not supporting `full_path` if it's too much.